src/metabase/lib/stage.cljc

```diff
@@ -149,8 +149,12 @@
    unique-name-fn :- fn?]
   (when card-id
     (when-let [card (lib.metadata/card query card-id)]
-      (lib.metadata.calculation/visible-columns query stage-number card {:unique-name-fn               unique-name-fn
-                                                                         :include-implicitly-joinable? false}))))
+      (->> (lib.metadata.calculation/visible-columns
+            query stage-number card {:unique-name-fn               unique-name-fn
+                                     :include-implicitly-joinable? false})
+           ;; questions should not have implicitly joinable columns (#30950)
+           (mapv #(dissoc % :fk-target-field-id))
+           not-empty))))
```

Comment:
I could not figure out why this works, maybe updating the comment would be helpful?

1. With `:include-implicitly-joinable? false` I would not expect any implicitly joinable columns, and am surprised the `dissoc` is necessary.
2. Doing this `dissoc` would seem to mark columns as not implicitly joinable but somehow this causes removal of implicitly joinable columns somewhere else.